numpy/conftest.py

```diff
@@ -136,3 +144,72 @@ def weak_promotion(request):
 
     yield request.param
     numpy._set_promotion_state(state)
+
+
+if HAVE_SCPDT:
```

Comment:
I don't really think (not quite sure) that this whole pattern is the pytest way to do things.

`warnings_errors_and_rng` looks like it could be a `hook` on the plugin side, most other configs can probably be part of `pyproject.toml`, although I suspect one can set them also in the `conftest.py` (even if `scipy-doctest` isn't available).

Reply:
The context manager technically could be a pytest hook. I would not do it though, since it's breaks the separation between the doctest and pytest layers. 
Setting them unconditionally is a tad strange: these are specific for `scipy-doctest`.